lib/dal/src/pkg.rs

```diff
@@ -252,20 +261,27 @@ where
     Key: Eq + PartialEq + std::hash::Hash,
 {
     pub fn new() -> Self {
-        let change_set_map = HashMap::new();
-
         Self {
-            inner: change_set_map,
+            inner: HashMap::new(),
+            overrides: HashMap::new(),
         }
     }
 
     pub fn get(&self, key: &Key) -> Option<&Thing> {
-        self.inner.get(key)
+        match self.overrides.get(key) {
+            Some(override_key) => self.inner.get(override_key),
+            None => self.inner.get(key),
+        }
     }
 
     pub fn insert(&mut self, key: Key, thing: Thing) -> Option<Thing> {
         self.inner.insert(key, thing)
     }
+
+    pub fn insert_override(&mut self, to_override_unique_id: Key, override_unique_id: Key) {
+        self.overrides
+            .insert(to_override_unique_id, override_unique_id);
+    }
```

Comment:
We also found that these were incorrectly inverted, though we believe that to have not been the core cause of the rever.